language/Type-Classes.md

```diff
@@ -80,27 +80,50 @@ In fact, a type similar to this `AddInt` is provided in `Data.Monoid.Additive`,
 
 For multi-parameter type classes, the orphan instance check requires that the instance is either in the same module as the class, or the same module as at least one of the types occurring in the instance. (TODO: example)
 
+## Kinds
+
+A typeclass's parameter can specify the Kind. If it doesn't specify the type's Kind, it can be inferred by methods in the class.
+
+``` purescript
+class A a
+-- `a` is kind `Type`
+class B (b :: Type)
+-- `b` is kind `Type`
+class C c
+  f :: forall a b. (a -> b) -> c a -> c b
+-- `c` is kind `Type -> Type`
+class D (d :: Type -> Type)
+-- `d` is kind `Type -> Type`
+class E (e :: # Type)
+-- `e` is kind `# Type`
+```
+
+If a typeclass's parameter's Kind is `Type`, an instance cannot be made for a type of kind `# Type` or `Type -> Type`.
+
 ## Rows
 
-Rows can't appear in instances as the head.
+Concrete Row literals cannot appear in instances.
 
 ``` purescript
 class M t
--- XXX Can't write an instance for a specific row.
+-- Can't write an instance for a specific row.
 instance mClosedRow :: M ()
+class M' (t :: # Type)
```

Comment:
No need to define two different classes in this example; just put the kind signature on `M` I think.